lib/CMakeLists.txt

```diff
@@ -255,17 +255,25 @@ check_c_source_compiles("#include <arm_neon.h>\nint main(){ uint8_t *dest; uint8
 	                neon_compile_result)
 
 if(neon_compile_result)
-    check_c_source_compiles("int main(){asm volatile(\"vrev32.8 q0, q0\");}"
+    check_c_source_compiles("#include <volk/volk_common.h>\n int main(){__VOLK_ASM volatile(\"vrev32.8 q0, q0\");}"
                             have_neonv7_result )
-    check_c_source_compiles("int main(){asm volatile(\"sub v1.4s,v1.4s,v1.4s\");}"
+    check_c_source_compiles("#include <volk/volk_common.h>\n int main(){__VOLK_ASM volatile(\"sub v1.4s,v1.4s,v1.4s\");}"
                             have_neonv8_result )
 
-    if (have_neonv7_result)
-        OVERRULE_ARCH(neonv8 "CPU is armv7")
+    # if both are supported we let VOLK choose the preferred one
+    if (NOT (have_neonv7_result EQUAL have_neonv8_result))
```

Comment:
Seems like the logic here should get cleaned up, otherwise I like what this PR does!

Reply:
Hi.
Sorry, but I'm unable to see which change I need to add. When neonv7 xor neonv8 is true (ie only one version supported) we need to unset the neonvX not supported no?
Or maybe it's the comment just before the if?

Reply:
According to @balister ARM CPUs may support both architectures.
I hope he can comment on this. Options I'm aware of are:
1. Test for NEON versions separately and only run `OVERRULE_ARCH` if particular version is not found. We may have both versions enabled.
2. Decide which arch we prefer. Test for this arch, if available prune the other arch. No further test necessary. If only fallback arch is available, use this one.
3. Something else.

In any case, we need to make a decision. I hope we can make an informed decision.

Reply:
So only one of those tests will succeed. I suggest to test for both archs and just call `OVERRULE_ARCH` on any arch where the test fails.

Now I wonder why we did this general NEON check first and then tried to deduce the version. I'd expect the code to check for each version and if it is not available just prune it from the list.

Can you put in a comment that clearly states, we expect only one of those archs to be present at any given time?

Are there any other implications?

I've read a bunch on NEON and ARM so far. Basically, we can have 2 states:
1. 32bit with aarch32 and neonv7
2. 64bit with aarch64 and neonv8

They exclude each other. It wouldn't be reasonable to try to compile for both archs. My conclusion is that we can just check for specific NEON versions and call `OVERRULE_ARCH` on that architecture if it is not available. I hope that simplifies this code.